maps/src/test/java/org/odk/collect/maps/layers/OfflineMapLayersImporterTest.kt

```diff
@@ -240,6 +244,43 @@ class OfflineMapLayersImporterTest {
         assertThat(booleanCaptor.secondValue, equalTo(false))
     }
 
+    @Test
+    fun `the warning dialog is displayed if some selected files are not supported and the importer dialog is kept displayed`() {
+        val file1 = TempFiles.createTempFile("layerA", ".txt")
+        val file2 = TempFiles.createTempFile("layerB", MbtilesFile.FILE_EXTENSION)
+
+        launchFragment().onFragment {
+            it.viewModel.loadLayersToImport(listOf(file1.toUri(), file2.toUri()), it.requireContext())
+
+            scheduler.flush()
+
+            val context = ApplicationProvider.getApplicationContext<Application>()
+            onView(withText(context.getLocalizedQuantityString(R.plurals.non_mbtiles_files_selected_title, 1, 1))).inRoot(isDialog()).check(matches(isDisplayed()))
+            onView(withText(R.string.some_non_mbtiles_files_selected_message)).inRoot(isDialog()).check(matches(isDisplayed()))
+            onView(withText(R.string.ok)).inRoot(isDialog()).check(matches(isDisplayed()))
+
+            assertThat(it.isVisible, equalTo(true))
+        }
+    }
+
+    @Test
+    fun `the warning dialog is displayed if all selected files are not supported and the importer dialog is dismissed`() {
+        val file = TempFiles.createTempFile("layerA", ".txt")
+
+        launchFragment().onFragment {
+            it.viewModel.loadLayersToImport(listOf(file.toUri()), it.requireContext())
+
+            scheduler.flush()
+
+            val context = ApplicationProvider.getApplicationContext<Application>()
+            onView(withText(context.getLocalizedQuantityString(R.plurals.non_mbtiles_files_selected_title, 1, 1))).inRoot(isDialog()).check(matches(isDisplayed()))
+            onView(withText(R.string.all_non_mbtiles_files_selected_message)).inRoot(isDialog()).check(matches(isDisplayed()))
```

Comment:
It'd be cleaner for these tests to use the helpers available in `Assertions`/`Interactions`, but I'm pretty sure that's something you were going to deal with in #6201 anyway.